src/taipy/gui/utils/date.py

```diff
@@ -30,7 +31,9 @@ def _date_to_ISO(date_val: t.Union[datetime, date, time]) -> str:
     return date_val.isoformat()
 
 
-def _ISO_to_date(date_str: str) -> datetime:
+def _ISO_to_date(date_str: str) -> t.Union[datetime, date]:
     # return datetime.strptime(date_str, '%Y-%m-%dT%H:%M:%S.%fZ')
     # return datetime.fromisoformat(date_str)
-    return parser.parse(date_str)
+    date = parser.parse(date_str)
+    date_regex = r"^[A-Z][a-z]{2} (?:Jan|Feb|Mar|Apr|May|Jun|Jul|Aug|Sep|Oct|Nov|Dec) \d{2} \d{4}$"
```

Comment:
what happens if the browser local has an impact on toDateString ?

Reply:
it should not according to mdm as it is compatible with all browsers

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/toDateString

Reply:
If input is ISO (as the function name infers) then this pattern should never occur

Reply:
Oh, do you mean that the browser timezone would affect the date parsing

Reply:
it could
and if you use YYY-MM-DD you can simply test the length

Reply:
right now, the toDateString method is the one being used when withTime = False so we dont have to concern with other cases such as YYYY-mm-dd. For the affect of tz on the date parsing, I trust that the date selector would do the job of getting the correct date :D